plugins/xml/core/src/test/java/org/pentaho/di/trans/steps/getxmldata/GetXMLDataStepAnalyzerTest.java

```diff
@@ -323,4 +323,9 @@ public void testGetXMLDataExternalResourceConsumer() throws Exception {
     assertEquals( GetXMLDataMeta.class, consumer.getMetaClass() );
   }
 
+  @Test
+  public void testNewInstance(){
```

Comment:
It would be good to add a test for cloneAnlayzer - to ensure that the clone isn't equal to the original, much like is being done here:
https://github.com/pentaho/pentaho-metaverse/pull/524/files#diff-196b64eefb5cedb6bdbcdcdf5048c586R34

Reply:
I didn't look closely, it not the metaverse project. my bad.

Reply:
Something like this... since you cannot extend the ClonableStepAnalzyerTest and reuse its common functionality:
 ```
 public void testCloneAnalyzer() {
    final GetXMLDataStepAnalyzer analyzer = new GetXMLDataStepAnalyzer();
    // verify that cloneAnalyzer returns an instance that is different from the original
    Assert.assertNotEquals( analyzer, analyzer.cloneAnalyzer() );
  }
```

Reply:
@NJtwentyone no worries - I actually envisioned this class being reusable everywhere, so it didn't occur to me until now that it wouldn't be possible